ooni/nettests/blocking/web_connectivity.py

```diff
@@ -173,9 +177,20 @@ def setUp(self):
         self.report['tcp_connect'] = []
         self.report['control'] = {}
 
-        self.hostname = urlparse(self.input).netloc
-        if not self.hostname:
-            raise AbsentHostname('No hostname', self.input)
+        parseResult = urlparse(self.input)
+
+	if parseResult.scheme == "":
+            url = "http://{}".format(self.input)
+        elif parseResult.scheme in ['http', 'https']:
+            url = self.input
+        else:
+            raise InvalidScheme("Invalid Scheme", parseResult.scheme)
```

Comment:
For consistency I would actually use the same code (maybe moving it into a function called `maybe_prepend_schema`) as the one in here: https://github.com/TheTorProject/ooni-probe/pull/760/files#diff-ff2446d60c29cb02ef8f4b286402a1dfR140. Otherwise we can have unexpected results depending on whether a URL is passed via the `--url` option or from a file.